woocommerce/woocommerce
# [API] - Add a visual endpoint for authenticating

As proposed in #6930 

Steps to test:

Need to send the follows parameters by GET or POST:
- `app_name` - APP name.
- `user_id` - User ID/Slug in the application and not from WordPress/WooCommerce
- `return_url` - URL to sent back the customer
- `scope` - Available scopes: `read`, `write` and `read_write`
- `callback_url` - URL over SSL that will receive the consumer data

And need to use `urlencode()` in each field. Example URL:

```
wc-auth/v1/authorize?app_name=My+App+Name&return_url=http%3A%2F%2Fclaudiosmweb.com&scope=write&callback_url=https%3A%2F%2Frequestb.in%2Fxzct6bxz&user_id=123
```

Example of consumer date:

``` json
{
  "user_id": "123", // User ID/SLug in the application and not from WordPress/WooCommerce
  "consumer_key": "ck_123456789",
  "consumer_secret": "cs_123456789",
  "key_permissions": "write"
}
```

Important to remember that callback url need SSL, so if test how something like http://requestb.in, need edit the `WC_Auth::post_consumer_data()` to not verify the SSL.
